conf/machine/iot2050.conf

```diff
@@ -10,11 +10,13 @@
 
 DISTRO_ARCH ?= "arm64"
 
-PREFERRED_PROVIDER_u-boot-${MACHINE} = "u-boot-iot2050"
+PREFERRED_PROVIDER_u-boot-${MACHINE} = "u-boot-iot2050-${IOT2050_VARIANTS}"
 
 KERNEL_NAME ?= "iot2050"
 
 IMAGE_TYPE ?= "wic-img"
 WKS_FILE ?= "iot2050.wks.in"
 
 IMAGE_INSTALL += "u-boot-script"
+
+IOT2050_VARIANTS ??= "pg1"
```

Comment:
`IOT2050_VARIANTS` is not needed. Default to one variant in this conf, likely pg2, and change PREFERRED_PROVIDER_u-boot-iot2050 in the kas option.

Reply:
Done

Reply:
```
local_conf_header:
  u-boot: |
    PREFERRED_PROVIDER_u-boot-${MACHINE} = "u-boot-iot2050-pg1"
```

Doesn't work for increment build, the `build/conf/local.conf` changed accordingly however isar doesn't hornor this change.

Reply:
Missing weak assignment in the distro.conf?

Reply:
Yep. Actually this comments has been deleted by me, seems the network didn't work as expected again.